posts/hardening-kubernetes-and-what-that-entails-with-entando.md

```diff
@@ -0,0 +1,103 @@
+
+# Shift Security Left
+
+As mentioned above, "shift security left" means thinking about security early and often during the software development life cycle (SDLC). There are several best practices to follow when building an application on Kubernetes and building the infrastructure that Kubernetes runs on.
+
+## Development, Build, and Infrastructure Best Practices
+
+### Entando and JHipster
+
+Entando relies on JHipster for code generation and project scaffolding.  Using code generators is a great way to ensure that your project has a structure and base setup that follows best practices. When developing applications, always follow your language of choice's best practices. Once a component is created, whether a microservice or a micro frontend, you will likely be containerizing it for deployment onto k8s. 
+
+### Base Image Vulnerabilities
+
+A strong foundation is key.  Base images are a common and well known threat vector that are a constant source of risk.  Entando provides hardened base images for Java and Postgres.  Their core component images are built on top of Redhat's ubi8 (universal base images).  These base images make for a quick pass regarding image scanning; they are minimal and have only the necessary software installed to perform their dedicated task.
+
+Entando comes with a handy command line tool that will help you containerize your "micro components".  Image scanning comes in handy at this point in the process. There is a load of different tools out there that can scan your docker images for common vulnerability enumerations (CVEs). A popular choice is "docker scan." This suite of scanning tools looks for known vulnerabilities within operating systems, packages, and languages.  Speaking of image scanning, let us talk about where image scanning should live.  In the pipeline of course!
+
+### Entando Pipelines
+
+Having secure containers is excellent, but if the pipelines they move through aren't secure, it will make little difference.  Entando has well documented CI/CD process that provide a GitFlow-like workflow.  These pipeline recommendations align with the same ethos we have seen thus far.  Add opinionated architecture and guardrails to minimize the number of human errors through the reduction of guesswork. More information can be found about Entando's CI/CD processes can be found [here](https://developer.entando.com/next/tutorials/create/pb/github-actions-workflow.html#overview).
```

Comment:
I think the previous version referenced the full Gitflow-like workflow which Entando Engineering uses internally - https://github.com/entando/entando-pipelines - based on strong opinions and git macros while the developer.entando.com reference is a simple github-actions starting point.  Maybe just refine the last sentence to this:
`More information can be found about Entando's CI/CD support [here](https://developer.entando.com/next/tutorials/create/pb/github-actions-workflow.html#overview) or Entando's own opinionated build pipelines [here](https://github.com/entando/entando-pipelines).
`